src/lib/y2storage/blk_device.rb

```diff
@@ -49,13 +52,21 @@ def encrypted?
       !encryption.nil?
     end
 
-    # Filesystem placed in the device, either directly or through an encryption
-    # layer.
+    # Filesystem in the device
+    #
+    # By default it returns a filesystem placed either directly on the device
+    # or through an encryption layer.
+    #
+    # It can be forced to return only the direct filesystem (no encryption or any
+    # other layer in between).
     #
-    # @return [Filesystems::BlkFilesystem] nil if neither the raw device or its
-    #   encrypted version are formatted
-    def plain_blk_filesystem
-      encrypted? ? encryption.blk_filesystem : blk_filesystem
+    # @param [traverse_encryption] if set to false, the method will return nil
+    #   if the device is encrypted, no matter if the encrypted device is
+    #   formatted or not.
+    # @return [Filesystems::BlkFilesystem] nil if the device is not formatted
+    def blk_filesystem(traverse_encryption: true)
```

Comment:
well, I do not like this name. It will break in future if there will be more devices that can be between filesystem and blk_device. so what about `traverse: true`?